assets/src/util/math.ts

```diff
@@ -7,3 +9,26 @@
  */
 export const clamp = (value: number, min: number, max: number): number =>
   Math.min(Math.max(min, value), max)
+
+/**
+ * Finds the element in {@link positions} that is closest to {@link point}
+ * @param positions List of coordinates to search
+ * @param point Point of reference to check distance to
+ * @returns The closest point in {@link positions} to {@link point}
+ */
+export const closestPosition = (
+  positions: LatLngExpression[],
+  point: LatLng
+) => {
+  const positionsByDistance = positions
+    .map((currentPosition, index) => ({
+      distance: point.distanceTo(currentPosition),
+      position: currentPosition,
+      index,
+    }))
+    .sort((lhs, rhs) => lhs.distance - rhs.distance)
+
+  return positionsByDistance[0] ?? undefined
+
+  // Interpolate here
```

Comment:
I didn't quite mean to leave this in, but also, I definitely think we'll need to do this at some point, because it'll otherwise feel weird if someone clicks on the line, and there isn't a point nearby, and it snaps to that point, instead of the point on the line.

Reply:
Agreed it's fine to leave a comment, but maybe expand on it a bit to explain that if we do want to interpolate in the future, this is where we'd do it (as it stands I can imagine someone being confused by seeing this comment and wondering where the interpolation logic is).

Reply:
I'm open to/would appreciate feedback on the wording if anything stands out, PTALA bbe5cb2839e837f93f93365779714be33f1179c0

Reply:
Looks good to me!

Reply:
> Do we have a ticket in the backlog to do this interpolation?

Not yet, because I figured we'd wait till it's a problem, but I can write one!